test/integration/file_artifact_test.js

```diff
@@ -228,4 +231,67 @@ suite('artifact extration tests', () => {
     // Missing artifact should have an error...
     assert.equal(result.artifacts['public/my-missing.txt'].storageType, 'error');
   });
+
+  test('upload retry', async () => {
+    let worker = new TestWorker(DockerWorker);
+    await worker.launch();
+
+    worker.on('task run', function() {
+      iptables.reject({
```

Comment:
you can also decide to just read to a buffer if the file is small than say 200 MB... most files probably are... We are already uploading sequentially anyways, as that is how docker gives us the files...